### PR TITLE
Update version info endpoint

### PR DESCRIPTION
**Pull request by viveksyngh**
This commits updates version endpoint to use ProviderInfo and VersionInfo structs from
faas-provider.GatewayInfo now uses ProviderInfo and VersionInfo structs. GatewayInfo is
left within gateway. Consumer of this endpoint should use typed struct GatewayInfo to
parse response.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## How Has This Been Tested?

I have tested this on my local with two different version of faas-netes ( one without change and other with change.

`http://gateway:8080/system/info` endpoint is working fine and giving expected response in both case.

```json
{"provider":{"provider":"faas-netes","version":{"sha":"713fff78bcff1f47b55f4ca20426d8912d0d0c4d","release":"dev"},"orchestration":"kubernetes"},"version":{"sha":"","release":"dev"},"arch":"amd64"}
```

older `faas-cli version` command is giving error due to update in struct 
```
faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  c12d57c39ac4cc6eef3c9bba2fb45113d882432f
 version: 0.12.14
Handling connection for 8080
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/openfaas/faas-cli/commands.getGatewayDetails(...)
	/private/tmp/faas-cli-20201009-42245-191fhgi/commands/version.go:157
github.com/openfaas/faas-cli/commands.printServerVersions(0x1593aa0, 0xc000130008)
	/private/tmp/faas-cli-20201009-42245-191fhgi/commands/version.go:112 +0x765
github.com/openfaas/faas-cli/commands.runVersionE(0x1807ac0, 0x1844898, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/faas-cli-20201009-42245-191fhgi/commands/version.go:64 +0x44a
github.com/spf13/cobra.(*Command).execute(0x1807ac0, 0x1844898, 0x0, 0x0, 0x1807ac0, 0x1844898)
	/private/tmp/faas-cli-20201009-42245-191fhgi/vendor/github.com/spf13/cobra/command.go:838 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x18087e0, 0x9, 0x159eb00, 0xc0001a5790)
	/private/tmp/faas-cli-20201009-42245-191fhgi/vendor/github.com/spf13/cobra/command.go:943 +0x336
github.com/spf13/cobra.(*Command).Execute(...)
	/private/tmp/faas-cli-20201009-42245-191fhgi/vendor/github.com/spf13/cobra/command.go:883
github.com/openfaas/faas-cli/commands.Execute(0xc000128aa0, 0x2, 0x2)
	/private/tmp/faas-cli-20201009-42245-191fhgi/commands/faas.go:78 +0xde
main.main()
	/private/tmp/faas-cli-20201009-42245-191fhgi/app.go:19 +0x7a
```

This will may break the older version of the CLI for `faas-cli version` command. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@LucasRoesler can you suggest a way to have the best of both here? No error in older versions of the CLI, and the new typed struct elsewhere? AFAIK the schema hasn't changed

**Comment by LucasRoesler** [comment]
@alexellis unfortunately, there is no good answer here.  Either the new api schema needs to include the previous fields (and the server needs to populate them) _or_ we need a new api version and we need to sever both v1 and v2 endpoints for some amount of time. 

The new data, moving it to new field names seems very undesirable. Adding a new endpoint and deprecating the old one is probably the simplest but has its own problem, what if the new CLI points at an old server? The CLI should know how to fallback to the original API if the new API is not found.  Basically a variation of API negotiation like the docker CLI does

File: gateway/handlers/infohandler.go [hunk 1]
```diff
@@ -24,31 +25,22 @@ func MakeInfoHandler(h http.Handler) http.HandlerFunc {
...
-			Version: &types.VersionInfo{
+			Version: &providerTypes.VersionInfo{
 				CommitMessage: version.GitCommitMessage,
 				Release:       version.BuildVersion(),
 				SHA:           version.GitCommitSHA,
 			},
-			Provider: &types.ProviderInfo{
-				Version:       providerVersion,
-				Name:          provider["provider"].(string),
-				Orchestration: provider["orchestration"].(string),
```
Comment (alexellis) [inline gateway/handlers/infohandler.go:49]: @viveksyngh why are we no longer setting orchestration here?

**Comment by alexellis** [comment]
Sorry for causing confusion here - the schema didn't change, it was another error we were bumping into caused by an empty string in gateways built during development.

https://github.com/openfaas/faas/commit/2e2250afe843e94fcc34e85c5c5173c8602ca56c

OK to merge.

**Review by alexellis — APPROVED** [review]
LGTM

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
@viveksyngh Doesn't the gateway use go mod now?

**Comment by viveksyngh** [comment]
No it does not .. I think we should migrate it to use go mod

**Comment by alexellis** [comment]
Your wish is my command. #1617 

Want to rebase?

**Review by alexellis — CHANGES_REQUESTED** [review]
Rebase required and go mod instead of dep (the original PR was fine, but I migrated us to modules between then and now)

**Comment by viveksyngh** [comment]
@alexellis  Rebased it.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**